solr/modules/hdfs/src/java/org/apache/solr/hdfs/update/HdfsUpdateLog.java

```diff
@@ -81,31 +77,15 @@ public void init(PluginInfo info) {
     log.info("Initializing HdfsUpdateLog: tlogDfsReplication={}", tlogDfsReplication);
   }
 
-  private Configuration getConf(Path path) {
-    Configuration conf = new Configuration();
-    if (confDir != null) {
-      HdfsUtil.addHdfsResources(conf, confDir);
-    }
-
-    String fsScheme = path.toUri().getScheme();
-    if (fsScheme != null) {
-      conf.setBoolean("fs." + fsScheme + ".impl.disable.cache", true);
-    }
-    return conf;
-  }
-
   @Override
-  public void init(UpdateHandler uhandler, SolrCore core) {
-
-    // ulogDir from CoreDescriptor overrides
-    String ulogDir = core.getCoreDescriptor().getUlogDir();
-
-    this.uhandler = uhandler;
-
+  protected boolean initCheckRepoen(SolrCore core) {
     synchronized (fsLock) {
       // just like dataDir, we do not allow
       // moving the tlog dir on reload
       if (fs == null) {
+        // ulogDir from CoreDescriptor overrides
+        String ulogDir = core.getCoreDescriptor().getUlogDir();
```

Comment:
(note I didn't write this code; I just moved these 2 lines)

I think the logic around dataDir / ulogDir resolution is pretty confusing; I'm not comfortable improving this TBH without a bigger investment it than my goal right now.  Note init() can be called additional times for core reloads so the dataDir will hold whatever it had from before.  It appears that's pertinent.

@magibney touched similar code recently; maybe would offer input

Reply:
There's some weird overlap between this PR and #1895 -- it's taking me a while to make sense of the relationships here, so thanks for bearing with me 😬 .

For all the work I've put into this area of the code recently, it's still quite confusing, and I'm hoping that this PR will help clarify things. I hesitate to suggest, but I'm wondering if it might be better to incorporate this refactor into #1895, since I'm not sure it makes sense to do a rote refactor of code that is iirc basically buggy. I'm taking a stab at what that would look like now, mainly to better understand the code.

I'll follow up soon.

Reply:
A _rote_ refactor is safe no matter if the code was buggy or not; no?  Any way, if you're doing different things here such that the two methods I pulled out are not needed (yes / no?) then I should not do this change at this time.  I saw code duplication and aimed to address it.  It's at the end of the init method; same code, albeit not char-for-char.  This PR isn't super important to me; no rush.

Reply:
>A rote refactor is safe no matter if the code was buggy or not; no?

💯 yeah, I wasn't suggesting it wasn't safe, and I think this PR was an improvement.

962b99f430c56f7387fbf233950fbc12946aaa70 applies the spirit of this PR to #1895, and I'm pretty happy with the results (it forced me to confront some of the duplication that had been a lingering source of confusion).